cpp/src/text/subword/subword_tokenize.cu

```diff
@@ -0,0 +1,292 @@
+  thrust::transform_exclusive_scan(
+    execpol->on(stream),
+    thrust::make_counting_iterator<cudf::size_type>(0),
+    thrust::make_counting_iterator<cudf::size_type>(strings_count + 1),
+    offsets_per_log.begin(),
+    [device_offsets, do_truncate, max_sequence_length, stride] __device__(cudf::size_type idx) {
+      uint32_t num_tokens = device_offsets[idx + 1] - device_offsets[idx];
+      if (do_truncate || num_tokens <= max_sequence_length) return uint32_t{1};
+      return 1 + ((num_tokens - max_sequence_length + stride - 1) / stride);
+    },
+    uint32_t{0},
+    thrust::plus<uint32_t>());
+  // last element is the total number of tokens
+  uint32_t const nrows_tensor_token_ids = offsets_per_log.element(strings_count, stream);
+
+  // compute global_row to log, and global_row to within_log_row correspondence
+  rmm::device_uvector<uint32_t> row2log(nrows_tensor_token_ids, stream);
+  auto d_row2log = row2log.data();
+  rmm::device_uvector<uint32_t> row2row_within_log(nrows_tensor_token_ids, stream);
+  auto d_row2row_within_log = row2row_within_log.data();
+  thrust::for_each_n(execpol->on(stream),
+                     thrust::make_counting_iterator<uint32_t>(0),
+                     strings_count,
+                     [d_offsets_per_log, d_row2log, d_row2row_within_log] __device__(auto idx) {
+                       uint32_t offset = d_offsets_per_log[idx];
+                       uint32_t nrows  = d_offsets_per_log[idx + 1] - offset;
+                       for (uint32_t jdx = 0; jdx < nrows; ++jdx) {
+                         d_row2log[jdx + offset]            = idx;
+                         d_row2row_within_log[jdx + offset] = jdx;
+                       }
+                     });
+
+  // create output data columns
+  auto tensor_token_ids = cudf::make_numeric_column(cudf::data_type{cudf::type_id::UINT32},
+                                                    nrows_tensor_token_ids * max_sequence_length,
+                                                    cudf::mask_state::UNALLOCATED,
+                                                    stream,
+                                                    mr);
+  auto tensor_attention_mask =
+    cudf::make_numeric_column(cudf::data_type{cudf::type_id::UINT32},
+                              nrows_tensor_token_ids * max_sequence_length,
+                              cudf::mask_state::UNALLOCATED,
+                              stream,
+                              mr);
+  auto tensor_metadata = cudf::make_numeric_column(cudf::data_type{cudf::type_id::UINT32},
+                                                   nrows_tensor_token_ids * 3,
+                                                   cudf::mask_state::UNALLOCATED,
+                                                   stream,
+                                                   mr);
+
+  // compute final-tensor, mask, and metadata
+  kernel_compute_tensor_metadata<<<nrows_tensor_token_ids, max_sequence_length, 0, stream>>>(
```

Comment:
I'm very concerned about using a public API parameter directly in a kernel execution configuration. Blocks should be a multiple of 32 threads in size. And if the user passes 0 they could get a very inscrutible CUDA error. 

Grid dimensions should be computed locally (preferably with `grid_1d`) and safety checked.